BedrockServer.cpp

```diff
@@ -27,14 +31,23 @@ void BedrockServer::acceptCommand(SQLiteCommand&& command, bool isNew) {
         SALERT("Blacklisting command (now have " << totalCount << " blacklisted commands): " << request.serialize());
     } else {
         unique_ptr<BedrockCommand> newCommand(nullptr);
-        if (SIEquals(command.request.methodLine, "BROADCAST_COMMAND")) {
+        if (SIEquals(command->request.methodLine, "BROADCAST_COMMAND")) {
             SData newRequest;
-            newRequest.deserialize(command.request.content);
+            newRequest.deserialize(command->request.content);
             newCommand = getCommandFromPlugins(move(newRequest));
             newCommand->initiatingClientID = -1;
             newCommand->initiatingPeerID = 0;
         } else {
-            newCommand = getCommandFromPlugins(move(command));
+            // If we already have an existing BedrockCommand (as opposed to a base class SQLiteCommand) this is
+            // something that we've already constructed (most likely, a response from leader), so no need to ask the
+            // plugins to build it over again from scratch (this also preserves the existing state of the command).
```

Comment:
This is going to look like the weirdest change in this PR, so I'll try to add some further explanation:

We call `server.acceptCommand` from a few different places. One of the most common is returning an escalated command from `SQLiteNode` after leader has finished the escalation. Because SQLiteNode was previously storing `SQLiteCommand`s, we'd need to re-convert these to the appropriate bedrock command by calling `getCommandFromPlugins`.

But now, as it stores the original command, we don't need to generate a `BedrockCommand` if we've already created one before escalation. We can just use the one we already have. We use the dynamic cast for that. If it succeeds, then we know this is a `BedrockCommand` and not just an `SQLiteCommand`, so we jsut re-use the command we already have. (we need to release it from the pointer to the base class and create a new pointer to the derived class).

On the other hand, if it's not a `BedrockCommand` but just a base-class `SQLiteCommand`, which happens, for instance, when leader's `SQLiteNode` is reading an escalated command and passing it to `BedrockServer` (this will be a base-class command constructed in `SQLiteNode` from the command request, and the server will need to figure out which plugin should handle it), the `dynamic_cast` to the derived class will fail, and we still call `getCommandFromPlugins`.

There's probably a cleaner way to do this, but it's very localized to this function now so I'm OK with this small bit of code.